src/persist-client/src/shard_id.proto

```diff
@@ -0,0 +1,16 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+package mz_persist_client.shard_id;
+
+message ProtoShardId {
```

Comment:
do we need this wrapper? a shard_id should always be round-tripped though a string, so it's just adding overhead

Reply:
Hmm, indeed, I can just replace it with `impl RustType<String> for ShardId` which would avoid this intermediate struct, but still keep the nice `into_proto()`/`into_rust()` interface.